### PR TITLE
Use of decimal for stock minimum months

### LINKED ISSUE #5876: Allow decimal figures for reorder interval and min months of stock
The Vanga team needs to be able to reorder on a weekly basis.  We currently use months as our basis.  It's possible to make this into weeks using a decimal, and the math should work.  However, the database needs to be modified and the inputs modified in both the stock settings and depot settings to allow for this.  See below:

image

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR is related to #5876

Close #5876

### CODE REVIEW

File: client/src/modules/depots/modals/depot.modal.html [hunk 1]
```diff
@@ -129,8 +129,7 @@
         FORM.LABELS.MIN_MONTHS_SECURITY_STOCK
       </label>
       <input name="min_months_security_stock" class="form-control"
-        ng-model="DepotModalCtrl.depot.min_months_security_stock" type="number" min="1" autocomplete="off" required
-        bh-integer>
+        ng-model="DepotModalCtrl.depot.min_months_security_stock" type="number" min="0" autocomplete="off" required>
```
Comment (jmcameron) [inline client/src/modules/depots/modals/depot.modal.html]: We should consider making min="0.00001" (or similar) to prevent acceptance of zero.

**Review by jmcameron — CHANGES_REQUESTED** [review]
I think you need to also do similar input form modifications on Stock > Stock Settings  (client/src/modules/stock/settings/stock-settings.html).

**Comment by jniles** [comment]
bors try

File: client/src/modules/stock/settings/stock-settings.html [hunk 2]
```diff
@@ -24,7 +24,7 @@
             </label>
             <input type="number" class="form-control" autocomplete="off" name="month_average_consumption"
               ng-model="StockSettingsCtrl.settings.month_average_consumption"
-              ng-change="StockSettingsCtrl.setMonthAverage(value)" min="1" bh-integer>
+              ng-change="StockSettingsCtrl.setMonthAverage(value)" min="0.00001">
```
Comment (jmcameron) [inline client/src/modules/stock/settings/stock-settings.html:27]: Not sure it matters, but should these include:  type="number" 
?

**Review by jmcameron — COMMENTED** [review]
The code changes look good (see my notes).

I tested this with bhima_test.  Changing the values in stock settings works fine.  

However when I tried it with the Depot editor, it seems like the values are set properly, but when I edit again, I see that the value "Default Purchase order interval (in months)" is still 0 (I set it to 0.25).  See:
image

File: client/src/modules/stock/settings/stock-settings.html (hunk 2 above)
Comment (jniles) [reply client/src/modules/stock/settings/stock-settings.html:27]: They do - you missed it.  It's up above.

File: server/models/schema.sql [hunk 3]
```diff
@@ -461,7 +461,7 @@ CREATE TABLE `depot` (
   `allow_exit_transfer` TINYINT(1) UNSIGNED NOT NULL DEFAULT 0,
   `allow_exit_loss` TINYINT(1) UNSIGNED NOT NULL DEFAULT 0,
   `location_uuid` BINARY(16) NULL,
-  `min_months_security_stock` SMALLINT(5) NOT NULL DEFAULT 2,
+  `min_months_security_stock` DECIMAL(19,4) NOT NULL DEFAULT 2,
   `parent_uuid` BINARY(16) NULL,
   `dhis2_uid` VARCHAR(150) DEFAULT NULL,
   `default_purchase_interval` SMALLINT(5) NOT NULL DEFAULT 0,
```
Comment (jniles) [inline server/models/schema.sql]: This needs to be DECIMAL as well.  That is the problem @jmcameron ran into.

Comment (mbayopanda) [reply server/models/schema.sql]: I had done changes only in migrate.js

**Review by jniles — APPROVED** [review]
This worked for me.

bors r+

**Pull request merged**